unit-tests/live/d400/test-hdr.py

```diff
@@ -0,0 +1,575 @@
+                frame_exposure = out_depth_frame.get_frame_metadata(rs.frame_metadata_value.actual_exposure)
+                frame_gain = out_depth_frame.get_frame_metadata(rs.frame_metadata_value.gain_level)
+                seq_id = out_depth_frame.get_frame_metadata(rs.frame_metadata_value.sequence_id)
+
+                if seq_id == 0:
+                    test.check(frame_exposure == first_exposure)
+                    test.check(frame_gain == first_gain)
+                else:
+                    test.check(frame_exposure == second_exposure)
+                    test.check(frame_gain == second_gain)
+
+        pipe.stop()
+        depth_sensor.set_option(rs.option.hdr_enabled, 0)  # disable hdr before next tests
+        device.hardware_reset()
+        time.sleep(5)
+
+
+# CHECKING HDR AFTER PIPE RESTART
+with test.closure("HDR Running - restart hdr at restream"):
+    device = test.find_first_device_or_exit()
+    depth_sensor = device.first_depth_sensor()
+    if depth_sensor and depth_sensor.supports(rs.option.hdr_enabled):
+        cfg = rs.config()
+        cfg.enable_stream(rs.stream.depth)
+        pipe = rs.pipeline()
+        pipe.start(cfg)
+        depth_sensor.set_option(rs.option.hdr_enabled, 1)
+        test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+        for i in range(10):
+            data = pipe.wait_for_frames()
+        test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+        pipe.stop()
+        pipe.start(cfg)
+        test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+        pipe.stop()
+        depth_sensor.set_option(rs.option.hdr_enabled, 0)  # disable hdr before next tests
+
+
+"""
+helper method
+checks that the frames resulting from the hdr_merge processing block are generated
+from frames streamed within this method (and not using old frames) - see RSDSO-17341.
+
+To check that, we check that the hdr counter is between depth_counter and depth_counter-2, as in any case, if we use
+new frames we should be in that range, example:
+depth # |  hdr #    
+1           1
+2           1 (merged frames 1+2)
+3           1 (no frame to merge with 3 as 4 didn't arrive yet)
+4           3 (merged frames 3+4)
+5           3 (no frame to merge with 5 as 6 didn't arrive yet)
+6           5 (merged frames 5+6)
+... (more frames, until a frame counter reset)
+1           1 (hdr # should like the value we reset to)
+2           1 (continues normally)
+"""
+def check_hdr_frame_counter(pipe, num_of_frames, merging_filter):
+    """
+    :return: False if some metadata could not be read - otherwise, True
+    """
+    prev_depth_counter = -1
+    for i in range(num_of_frames):
+        data = pipe.wait_for_frames()
+
+        # get depth frame data
+        depth_frame = data.get_depth_frame()
+        if not test.check(depth_frame.supports_frame_metadata(rs.frame_metadata_value.sequence_id)):
+            return
```

Comment:
Could you add here some log reading the fact that this metadata could not be read?
Do you think it should fail the test? (I am not sure....)

Reply:
on a second look... I dont think we need sequence id here, right? just the frame counter, what if we just remove it and remove also `test.check(merged_frame.supports_frame_metadata(rs.frame_metadata_value.sequence_id)) `?